app/models/validated_pb_core.rb

```diff
@@ -32,4 +33,25 @@ def method_validate
     return if errors.empty?
     fail 'Method validation errors: ' + errors.join("\n")
   end
+  
+  def url_validate
+    errors = []
+    PBCore.instance_methods(false).grep(/(src|url)s?/).each do |method|
+      urls = [send(method)].select {|u| u}.flatten
+      urls.each do |url|
+        begin
+          Curl::Easy.new(url).tap do |curl|
+            curl.http_head
+            unless curl.response_code == 200
+              errors << "HEAD #{url} (from ##{method}) not 200: #{curl.status}"
+            end
+          end
+        rescue => e
+          errors << (["'##{method}' failed: #{e.message}"] + e.backtrace[0..2]).join("\n")
+        end 
+      end
+    end
+    return if errors.empty?
+    fail 'Method validation errors: ' + errors.join("\n")
+  end
```

Comment:
I'll add a hard-coded list as a double check, but I think it's good to also have it dynamic: if a new field of this type gets added, it would be good for it also to be checked.